addons/udes_stock/models/stock_move.py

```diff
@@ -462,16 +464,37 @@ def update_orig_ids(self, origin_moves):
                 prod_qty_key = (origin_move.product_id.id, origin_move.product_uom_qty)
                 free_origin_moves_by_prod_qty[prod_qty_key] = origin_move
 
-            for move, prod_qty in unlinked_qty_by_move.items():
-                try:
-                    new_origin_move = free_origin_moves_by_prod_qty.pop(
-                        (move.product_id.id, prod_qty)
-                    )
-                    orig_moves_by_move[move] |= new_origin_move
-                except KeyError:
-                    _logger.warning(
-                        "Unable to find origin move for %sx%s", move.product_id.name, prod_qty
+            if len(unlinked_qty_by_move) == 1:
+                # Special case: total unused origin quantities match the unlinked quantity.
+                move, prod_qty = list(unlinked_qty_by_move.items())[0]
+                rounding = move.product_id.uom_id.rounding
+                unused_qty = math.fsum(
+                    origin_moves.filtered(
+                        lambda m: m.product_id == move.product_id and m not in used_origin_moves
+                    ).mapped("product_uom_qty")
+                )
+                if float_compare(prod_qty, unused_qty, precision_rounding=rounding) == 0:
+
+                    unused_origs = origin_moves.filtered(
```

Comment:
nit: could move `unused_origs` above `unused_qty` then reference `unused_origs` to avoid writing `origin_moves.filtered(...` twice

Reply:
Good spot, fixed.